terraform/modules/spack/gitlab_runner_iam.tf

```diff
@@ -0,0 +1,125 @@
+locals {
+  # TODO: is it possible to get this from the gitlab provider?
+  gitlab_domain = "gitlab${var.deployment_name == "prod" ? "" : ".staging"}.spack.io"
+
+  mirror_roles = {
```

Comment:
The conditions here should match these: https://github.com/spack/spack/blob/aa9eb331080a3edeb876dd80552bb59243c69783/share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml#L71-L109 using the more limited IAM string matching. @scottwittenburg should probably also review these.